pxd.c

```diff
@@ -741,6 +826,19 @@ ssize_t pxd_add(struct fuse_conn *fc, struct pxd_add_ext_out *add)
 		goto out_module;
 	}
 
+	// if device already exists, then return it
+	pxd_dev = find_pxd_device(ctx, add->dev_id);
+	if (pxd_dev) {
+		module_put(THIS_MODULE);
+
+		if (add->enable_fp) {
+			__pxd_update_path(pxd_dev, &add->paths);
+		} else {
+			disableFastPath(pxd_dev, false);
```

Comment:
This can be a long running op. Since pxd_add is a synchronous call from px-storage, if px dies we would end up in D state. For any such we need to look at making them async.

Reply:
This is an IOCtl call from px-storage, px-storage failing will not result in D state, unless userspace IOs are serviced. If that were to happen px device is already in native path servicing IOs in native path and call to disableFastpath is a nop.
Only if the px device is in fastpath, will disable be called inline in an IOCtl call, which is okay.